File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/channels/ChannelsRepliesRequest.java
package com.github.seratch.jslack.api.methods.request.channels;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class ChannelsRepliesRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `channels:history`
     */
    private String token;

    /**
     * Channel to fetch thread from
     */
    private String channel;

    /**
     * Unique identifier of a thread's parent message
     */
    private String threadTs;

    ChannelsRepliesRequest(String token, String channel, String threadTs) {
        this.token = token;
        this.channel = channel;
        this.threadTs = threadTs;
    }

    public static ChannelsRepliesRequestBuilder builder() {
        return new ChannelsRepliesRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getChannel() {
        return this.channel;
    }

    public String getThreadTs() {
        return this.threadTs;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public void setThreadTs(String threadTs) {
        this.threadTs = threadTs;
    }

    public static class ChannelsRepliesRequestBuilder {
        private String token;
        private String channel;
        private String threadTs;

        ChannelsRepliesRequestBuilder() {
        }

        public ChannelsRepliesRequest.ChannelsRepliesRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public ChannelsRepliesRequest.ChannelsRepliesRequestBuilder channel(String channel) {
            this.channel = channel;
            return this;
        }

        public ChannelsRepliesRequest.ChannelsRepliesRequestBuilder threadTs(String threadTs) {
            this.threadTs = threadTs;
            return this;
        }

        public ChannelsRepliesRequest build() {
            return new ChannelsRepliesRequest(token, channel, threadTs);
        }

        public String toString() {
            return "ChannelsRepliesRequest.ChannelsRepliesRequestBuilder(token=" + this.token + ", channel=" + this.channel + ", threadTs=" + this.threadTs + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/channels/ChannelsRepliesRequest.kt
package work.socialhub.kslack.api.methods.request.channels

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class ChannelsRepliesRequest(
    /** Authentication token. Requires scope: `channels:history */
    override var token: String?,
    /** Channel to fetch thread from */
    var channel: String?,
    /** Unique identifier of a thread's parent message */
    var threadTs: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("channel", channel)
            it.addParam("thread_ts", threadTs)
        }
    }
}